src/software/ai/navigator/path_planner/path_planner_performance_test.cpp

```diff
@@ -0,0 +1,224 @@
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_10,
+     .num_iterations = 10},
+
+    {.name           = "20 obstacles on full field",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_20,
+     .num_iterations = 10},
+
+    {.name           = "30 obstacles on full field",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_30,
+     .num_iterations = 10},
+};
+
+template <typename PlannerT>
+std::pair<std::string, PathPlannerConstructor> name_and_constructor()
+{
+    return std::pair<std::string, PathPlannerConstructor>(
+        typeid(PlannerT).name(), []() { return std::make_unique<PlannerT>(); });
+}
+
+std::vector<std::pair<std::string, PathPlannerConstructor>>
+    path_planner_names_and_constructors = {
+        // add path planner constructors here
+        name_and_constructor<ThetaStarPathPlanner>(),
+        name_and_constructor<StraightLinePathPlanner>()};
+
+class PlannerPerformanceTest
+    : public testing::TestWithParam<
+          std::tuple<std::pair<std::string, PathPlannerConstructor>, PlannerTestCase>>
+{
+};
+
+// This test is disabled, it can be enabled by removing "DISABLED_" from the test name
+TEST_P(PlannerPerformanceTest, DISABLED_path_planner_performance)
+{
+    std::string name                     = std::get<0>(GetParam()).first.substr(2);
+    std::unique_ptr<PathPlanner> planner = std::get<0>(GetParam()).second();
+    auto planner_test_case               = std::get<1>(GetParam());
+
+    auto start_time = std::chrono::high_resolution_clock::now();
+    for (unsigned int i = 0; i < planner_test_case.num_iterations; i++)
+    {
+        planner->findPath(planner_test_case.start, planner_test_case.end,
+                          planner_test_case.navigable_area, planner_test_case.obstacles);
+    }
+    auto end_time = std::chrono::high_resolution_clock::now();
+
+    double duration_us = static_cast<double>(
+        std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time)
+            .count());
+    double avg_us = duration_us / (static_cast<double>(planner_test_case.num_iterations));
+
+    std::cout << std::endl
+              << name << " | # iterations = " << planner_test_case.num_iterations
+              << " | # obstacles = " << planner_test_case.obstacles.size()
+              << " | area = " << planner_test_case.navigable_area.area()
+              << " | path length = "
+              << (planner_test_case.start - planner_test_case.end).length() << std::endl;
+
+    std::cout << "Total time = " << duration_us / 1000.0
```

Comment:
nit: if we're printing in `ms` why not convert directly to `ms` with the `duration_cast` instead of `microseconds`?

Reply:
it's for better precision. I cleaned it up and went to nanosecond precision